### PR TITLE
Fuzz utils

### LINKED ISSUE #16: Add java client
Adding teku/artemis

### PR DESCRIPTION
**Pull request by gnattishness**
Fuzzing harnesses suitable for use with [beacon_fuzz](https://github.com/sigp/beacon-fuzz)

## PR Description

Implements harnesses that exercise relevant state transition functions.
They read SSZ-encoded bytes (generally a `BeaconState` + `Operation`, perform the operation, then output the resulting SSZ-encoded `BeaconState`).

## Fixed Issue(s)

Java-side of the implementation that fixes sigp/beacon-fuzz#16

## TODO

- Full integration testing (when `beacon-fuzz` supports v0.10.1)
- Ability to disable BLS signature verification (could be a separate PR)
- Squash and neaten commits

### CODE REVIEW

**Comment by gnattishness** [comment]
Currently builds, but quite likely some runtime failures.

**Comment by gnattishness** [comment]
Ready for review.

**Comment by gnattishness** [comment]
Updated to incorporate updates to master.
Additional testing and equality methods added by @mbaxter Thanks!

If this all looks fine, I'll give it a final rebase and squash

**Comment by gnattishness** [comment]
Squished into a single reasonably pretty commit.
Please let me know if there's any better attribution I can do protect against erasing the history of other's contributions!

**Comment by mbaxter** [comment]
> Squished into a single reasonably pretty commit.
> Please let me know if there's any better attribution I can do protect against erasing the history of other's contributions!

Thanks @gnattishness  - looks good!  I'll try to do a final review today 🎉

File: fuzz/src/main/java/tech/pegasys/teku/fuzz/FuzzUtil.java [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+import tech.pegasys.teku.fuzz.input.BlockFuzzInput;
+import tech.pegasys.teku.fuzz.input.BlockHeaderFuzzInput;
+import tech.pegasys.teku.fuzz.input.DepositFuzzInput;
+import tech.pegasys.teku.fuzz.input.ProposerSlashingFuzzInput;
+import tech.pegasys.teku.fuzz.input.VoluntaryExitFuzzInput;
+import tech.pegasys.teku.ssz.SSZTypes.SSZList;
+import tech.pegasys.teku.ssz.sos.ReflectionInformation;
+import tech.pegasys.teku.util.config.Constants;
+
+// TODO a Java FuzzHarness interface? - that way type safety can be checked at compile time
```
Comment (mbaxter) [inline fuzz/src/main/java/tech/pegasys/teku/fuzz/FuzzUtil.java]: Would you mind clearing out some of these TODO comments?  We should either implement the TODO's now or cut them.

**Review by mbaxter — COMMENTED** [review]
Hey @gnattishness - this looks good to me!  Just wondering if you can clean up some of the TODOs before we merge.

File: fuzz/src/main/java/tech/pegasys/teku/fuzz/FuzzUtil.java (hunk 1 above)
Comment (gnattishness) [reply fuzz/src/main/java/tech/pegasys/teku/fuzz/FuzzUtil.java]: Done! Sorry to leave them sitting around.

**Review by mbaxter — APPROVED** [review]

File: fuzz/src/main/java/tech/pegasys/teku/fuzz/FuzzUtil.java (hunk 1 above)
Comment (mbaxter) [reply fuzz/src/main/java/tech/pegasys/teku/fuzz/FuzzUtil.java]: np - thanks for the clean up!

**Pull request merged**